packages/client/tests/functional/adapter-errors/tests.ts

```diff
@@ -0,0 +1,87 @@
+import { ok } from '@prisma/driver-adapter-utils'
+
+import { NewPrismaClient } from '../_utils/types'
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let newPrismaClient: NewPrismaClient<typeof PrismaClient>
+
+testMatrix.setupTestSuite(
+  () => {
+    let prisma: PrismaClient
+
+    const queryRawError = new Error('Error in queryRaw')
+    const executeRawError = new Error('Error in executeRaw')
+    const transactionError = new Error('Error in startTransaction')
+    beforeAll(() => {
+      prisma = newPrismaClient({
+        adapter: {
+          flavour: 'sqlite',
+          queryRaw() {
+            throw queryRawError
+          },
+          executeRaw() {
+            throw executeRawError
+          },
+          startTransaction() {
+            throw transactionError
+          },
+          close() {
+            // method is not used at the moment
+            return Promise.resolve(ok(undefined))
+          },
```

Comment:
There's a mock adapter in `tests/functional/driver-adapters/_utils/mock-adapter.ts` but it currently throws the same error in each method, I think it would be better to update it to throw different errors like you did here and reuse it.

Reply:
also, would it make sense to move these test suites to `test/functional/driver-adapters`?

Reply:
I've added an export for all the errors mock adapter throws and re-used it here now.
Also, moved the tests.